core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -102,15 +103,28 @@ public XContentBuilder(XContent xContent, OutputStream bos) throws IOException {
     }
 
     /**
-     * Constructs a new builder using the provided xcontent, an OutputStream and some filters. The
-     * filters are used to filter fields that won't be written to the OutputStream. Make sure
-     * to call {@link #close()} when the builder is done with.
+     * Constructs a new builder using the provided xcontent, an OutputStream and
+     * some filters. If filters are specified, only those values matching a
+     * filter will be written to the output stream. Make sure to call
+     * {@link #close()} when the builder is done with.
      */
     public XContentBuilder(XContent xContent, OutputStream bos, String[] filters) throws IOException {
         this.bos = bos;
         this.generator = xContent.createGenerator(bos, filters);
     }
 
+    /**
+     * Constructs a new builder using the provided xcontent, an OutputStream and
+     * some filters. If filters are specified and inclusiveFilters is true, only
+     * those values matching a filter will be written to the output stream. If
+     * inclusiveFilters is false, those matching will be excluded. Make sure to
+     * call {@link #close()} when the builder is done with.
+     */
+    public XContentBuilder(XContent xContent, OutputStream bos, boolean inclusiveFilters, String[] filters) throws IOException {
```

Comment:
This constructor should be used in `XContentBuilder(XContent xContent, OutputStream bos, String[] filters)` with inclusive set to true